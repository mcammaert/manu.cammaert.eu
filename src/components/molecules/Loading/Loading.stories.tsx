import React from 'react';

import { storiesOf } from '@storybook/react';

import Loading from './Loading';

storiesOf('molecules/Loading', module).add('Loading', () => {
  return <Loading />;
});
