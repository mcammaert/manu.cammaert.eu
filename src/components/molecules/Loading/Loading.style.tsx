import styled from 'styled-components';

export const Loading = styled.div`
  position: absolute;
`;

export default { Loading };
