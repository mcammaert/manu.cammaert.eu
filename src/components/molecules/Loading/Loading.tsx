import React from 'react';

import { LoadingProps } from './Loading.types';

import S from './Loading.style';

const Loading: React.FC<LoadingProps> = () => (
  <S.Loading>
    <p>Laden...</p>
  </S.Loading>
);

export default Loading;
