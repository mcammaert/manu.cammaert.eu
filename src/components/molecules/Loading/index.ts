import * as S from './Loading.style';

export { default as Loading } from './Loading';
export * from './Loading.types';
export { S };
