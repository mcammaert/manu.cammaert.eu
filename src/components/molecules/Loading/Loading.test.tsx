import React from 'react';

import { cleanup } from '@testing-library/react';
import 'jest-styled-components';

import { renderComponent } from 'testing/react-testing-library';

import Loading from './Loading';

afterEach(cleanup);

const theme = {
  colors: {},
};

describe('Loading', () => {
  describe('default props', () => {
    it('should match the snapshot', () => {
      const component = <Loading />;
      const { container } = renderComponent({ theme, component });
      expect(container).toMatchSnapshot();
    });
  });
});
