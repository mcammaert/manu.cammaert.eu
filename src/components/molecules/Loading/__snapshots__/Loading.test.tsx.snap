// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading default props should match the snapshot 1`] = `
.c0 {
  position: absolute;
}

<div>
  <div
    class="c0"
  >
    <p>
      Laden...
    </p>
  </div>
</div>
`;
